guides/notebooks/native-navigation.livemd

```diff
@@ -0,0 +1,377 @@
+    </NavigationLink>
+    """
+  end
+
+  def render(assigns) do
+    ~H"""
+    <.link navigate={"/"}>To Main</.link>
+    """
+  end
+end
+|> KinoLiveViewNative.register("/about", ":index")
+
+import KinoLiveViewNative.Livebook, only: []
+import Kernel
+:ok
+```
+
+The `destination` attribute works the same as the `navigate` attribute on the web. The current LiveView will shut down, and a new one will mount without reloading the whole page or re-establishing a new socket connection.
+
+## Routing
+
+The `KinoLiveViewNative` smart cells used in this guide automatically define routes for us. Be aware there is no difference between how we define routes for LiveView or LiveView Native.
+
+The routes for the main and about pages might look like the following in the router:
+
+<!-- livebook:{"force_markdown":true} -->
+
+```elixir
+live "/", Server.MainLive
+live "/about", Server.AboutLive
+```
+
+## Native Navigation Events
+
+There are three main navigation functions in Phoenix LiveView. Each function sends a navigation event to the LiveView Native [LiveViewCoordinator](https://github.com/liveview-native/liveview-client-swiftui/blob/0e0fc6bbe5e95ef308e51551af0889acb09b87b3/Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift) through a channel.
+
+* redirect -> sends the [`redirect` event](https://github.com/liveview-native/liveview-client-swiftui/blob/9895c3b16d84a2683dcb1f127994be6c1bdf4919/Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift#L274-L281).
+* push_navigate -> sends a [`live_redirect` event](https://github.com/liveview-native/liveview-client-swiftui/blob/9895c3b16d84a2683dcb1f127994be6c1bdf4919/Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift#L258-L265).
+* push_patch -> sends a [`live_patch` event](https://github.com/liveview-native/liveview-client-swiftui/blob/9895c3b16d84a2683dcb1f127994be6c1bdf4919/Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift#L266-L273).
+
+The `LiveViewCoordinator` [sends those navigation requests to the LiveSessionCoordinator](https://github.com/liveview-native/liveview-client-swiftui/blob/0e0fc6bbe5e95ef308e51551af0889acb09b87b3/Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift#L353) which handles them natively.
+
+The LiveView Native [LiveSessionCoordinator](https://github.com/liveview-native/liveview-client-swiftui/blob/0e0fc6bbe5e95ef308e51551af0889acb09b87b3/Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift) manages the current [navigation path](https://github.com/liveview-native/liveview-client-swiftui/blob/0e0fc6bbe5e95ef308e51551af0889acb09b87b3/Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift#L43). The navigation path contains a list of [LiveNavigationEntries](https://github.com/liveview-native/liveview-client-swiftui/blob/0e0fc6bbe5e95ef308e51551af0889acb09b87b3/Sources/LiveViewNative/Coordinators/LiveNavigationEntry.swift) which acts as the history of pages the user has viewed.
```

Comment:
This section refers to implementation details of the client. `liveview-native-core` will likely handle navigation in the near future and this will become out of date. So it could be beneficial to omit this level of detail.

Reply:
agreed